Daemon/callback_storage.c

```diff
@@ -73,10 +78,32 @@ int StorageGetCallbacks(const char* channel, Callback** callbacks, int *number_o
         return -1;
 
     int cnt = 0;
-    for (CallbackList elem = root; elem; elem = elem->next)
+    for (CallbackNode* elem = root; elem; elem = elem->next)
         if (IsPrefixOf(elem->channel, channel))
             (*callbacks)[cnt++] = elem->callback;
 
     return 0;
 }
 
+int CallbackDeleteForNonexistentPID()
+{
+    // mai intai elimin toate non-root elements
+    if (root) {
+        CallbackNode* curr = root->next, *previous = root;
+        while (curr) {
+            if (!isAlivePID(curr->callback.PID)) {
+                printf(" --- Deleted (dead) callback PID/token:   %d | %d", 
+                       curr->callback.PID, curr->callback.token);
+                previous->next = curr->next;
+                free(curr->channel);
+                free(curr);
+                curr = previous->next;
+            }
+            else {
+                previous = curr;
+                curr = curr->next;
+            }
+        }
```

Comment:
What about the root?

Reply:
Lmao, you're right. I guess i just got carried away